mapiproxy/servers/default/emsmdb/emsmdbp_object.c

```diff
@@ -3138,6 +3138,117 @@ _PUBLIC_ struct emsmdbp_object *emsmdbp_object_ftcontext_init(TALLOC_CTX *mem_ct
 }
 
 /**
+   \details Store the target recipients attribute in out parameter.
+
+   \param emsmdbp_ctx pointer to the emsmdb provider cotnext
+   \param sharing_object the sharing message object to get the XML data from
+   \param [out] pointer to string where the TargetRecipients XML attribute is stored
+   \param mem_ctx pointer to memory context where the target_recipients_attr is created
+
+   \note On success target_recipients_attr string is allocated in
+   mem_ctx. It is up to the developer to free that memory context when
+   it is not needed anymore.
+
+   \return MAPI_E_SUCCESS on success, a mapi error code otherwise
+ */
+static enum MAPISTATUS emsmdbp_object_sharing_metadata_recipients_attr(struct emsmdbp_context *emsmdbp_ctx, struct emsmdbp_object *sharing_object, char **target_recipients_attr, TALLOC_CTX *mem_ctx)
+{
+	char			       *xml = NULL;
+	struct emsmdbp_prop_index      prop_index;
+	int			       i;
+	struct mapistore_message       *msg_data = NULL;
+	TALLOC_CTX		       *local_mem_ctx;
+	uint32_t		       contextID;
+
+	local_mem_ctx = talloc_named(NULL, 0, "sharing_metadata_recipients_attr");
+	OPENCHANGE_RETVAL_IF(!local_mem_ctx, MAPI_E_NOT_ENOUGH_MEMORY, NULL);
+
+	contextID = emsmdbp_get_contextID(sharing_object);
+	mapistore_message_get_message_data(emsmdbp_ctx->mstore_ctx, contextID, sharing_object->backend_object, local_mem_ctx, &msg_data);
+	if (msg_data && msg_data->recipients_count > 0) {
+		xml = talloc_asprintf(local_mem_ctx, "TargetRecipients=\"");
+		OPENCHANGE_RETVAL_IF(!xml, MAPI_E_NOT_ENOUGH_MEMORY, local_mem_ctx);
+
+		emsmdbp_fill_prop_index(&prop_index, msg_data->columns);
+		for (i = 0; i < msg_data->recipients_count; i++) {
+			if (prop_index.email_address != (uint32_t) -1) {
```

Comment:
See `emsmdbp_fill_prop_index` what is set in case of error. If I change this here to `0xFFFFFFFF`, then I have to change the assignment in that function too...